### ISSUE TITLE
Did this package work well for Mac OS 10.8.2 GeForce 660M GPU?

### ISSUE THREAD

**Issue by odieXin** [nitishsrivastava/deepnet#44]
I got libcudamat.dylib and libcudamat_conv.dylib,
then when I tried to test it with "python test_cudamat.py", 
the result is: 
Ran 0 tests in 0.000s
OK

Then, I tried to run sh runall.sh in the rbm folder and finally, it showed:

No GPU board available.

Could someone help figure out the issue?

Thanks so much!!!

**Comment by ansyral** [comment]
also has the problem on ubuntu10.04 with the Geforce gtx650 graphic card.is it the reason of the graphic card ?

**Comment by sanggeol** [comment]
I am running this package on OS X 10.9.1 with GeForce GT 650M, but it requires a little bit of modification.
As you might know, deepnet packages uses cudamat for gpu computing, and cudamat has a rather weird way of locking a gpu. ('cudamat/gnu_lock.py') It looks for /dev/nvidia-0 and create a symlink to it, which means that it will only work on Linux systems as macs don't have /dev/anything. 

So I have an even uglier solution. Go to /deepnet/trainer.py, and modify LockGPU. Mine looks like this.

board = 0
cm.cuda_set_device(board)
cm.cublas_init()
return board

Basically I'm skipping gpu locking and telling it to use gpu number 0 (which is the only gpu installed in my case). This is NOT a real solution, but it works for now. I would appreciate if anyone could share a better solution.

**Comment by danbri** [comment]
I hit this issue too on a mac, and sanggeol's hack works for me.
